### PR TITLE
Fix order of operations in failing transpose test

### PR DESCRIPTION
**Pull request by hsharsha**

### CODE REVIEW

File: xla/service/gpu/fusions/transpose_test.cc [hunk 1]
```diff
@@ -80,7 +80,7 @@ TEST_F(TransposeTest, ThreadIndexing021) {
         (d0, d1, d2, d3, d4, d5)[s0, s1, s2] -> (
           d3 floordiv 2,
           d0 floordiv 32 + s1 * 4,
-          (d3 mod 2) * 32 + d0 mod 32
+          d0 mod 32 + (d3 mod 2) * 32
```
Comment (hsharsha) [inline xla/service/gpu/fusions/transpose_test.cc:83]: Does the order of addition affect performance?

File: xla/service/gpu/fusions/transpose_test.cc [hunk 2]
```diff
@@ -79,9 +79,13 @@ TEST_F(TransposeTest, ThreadIndexing021) {
...
           d3 floordiv 2,
-          d0 floordiv 32 + s1 * 4,
-          (d3 mod 2) * 32 + d0 mod 32
-        )
+          d0 floordiv 32 + s1 * 4,)"
+#ifdef GOOGLE_CUDA
+          R"((d3 mod 2) * 32 + d0 mod 32)"
+#elif TENSORFLOW_USE_ROCM
+          R"(d0 mod 32 + (d3 mod 2) * 32)"
+#endif
```
Comment (olegshyshkov) [inline xla/service/gpu/fusions/transpose_test.cc:87]: Do you understand what's different in ROCm codegen that causes this difference? The expressions are equivalent, so something the construction is not deterministic. I would rather fix that inside indexing maps.

**Comment by kamaljeeti** [comment]
Hi @hsharsha , can you look into the comments? Thanks.

**Comment by hsharsha** [comment]
> Hi @hsharsha , can you look into the comments? Thanks.

Looks like this is similar to https://github.com/openxla/xla/pull/11567/ Will use the same pattern matching once that PR is accepted

**Comment by dimitar-asenov** [comment]
Hi @hsharsha It seems that #11567 is merged. Please take a look and adjust the current PR when you have the time. Thanks.

**Comment by hsharsha** [comment]
I think there are changes made to indexing maps and string matching ir. I will close this PR as the test is passing now and will revisit if there is a failure in the future.

**Comment by hsharsha** [comment]
I think there are changes made to indexing maps and string matching ir. I will close this PR as the test is passing now and will revisit if there is a failure in the future.

**Pull request closed without merging**